TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Controllers/TicketBookingController.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using TicketBookingAPI.Model;
+using TicketBookingAPI.Services;
+
+namespace TicketBookingAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class TicketBookingController : Controller
+    {
+        private CityDetailsService _service;
+        public TicketBookingController(CityDetailsService service) 
+        { 
+            _service= service;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<CityDetailsModel>>> GetAllCities() 
+        {
+            try 
+            {
+                var cities = await _service.GetAllCities();
+                return Ok(cities);
+            }catch(Exception ex) 
+            { 
+                return BadRequest(ex.Message);
+            }
+        }
+
+        [HttpGet("GetAllCitiesLike/{pattern}")]
```

Comment:
Have only one endpoint with string?, no need to have 2 different endpoints which are doing the same things